indico/util/date_time.py

```diff
@@ -448,13 +450,19 @@ def strftime_all_years(dt, fmt):
         return dt.replace(year=1900).strftime(fmt.replace('%Y', '%%Y')).replace('%Y', str(dt.year))
 
 
+@memoize_request
 def get_display_tz(obj=None, as_timezone=False):
-    display_tz = session.timezone if has_request_context() else 'LOCAL'
+    display_tz = session_tz = session.timezone if has_request_context() else 'LOCAL'
     if display_tz == 'LOCAL':
         if obj is None:
             display_tz = config.DEFAULT_TIMEZONE
         else:
             display_tz = getattr(obj, 'timezone', 'UTC')
     if not display_tz:
         display_tz = config.DEFAULT_TIMEZONE
-    return pytz.timezone(display_tz) if as_timezone else display_tz
+    try:
+        return pytz.timezone(display_tz) if as_timezone else display_tz
```

Comment:
when `as_timezone == False` do we want to return a potentially invalid timezone? Though I didn't find any place where we pass `as_timezone=False` but it still might be nice to be consistent?

Reply:
in principle there shouldn't be any invalid timezones anyway ;)

my main goal here  was to avoid failing in an ugly way - i already removed the crappy timezones from user data in the DB, but an existing session could still have a bad timezone and we have no good way to clear that.

but yeah, I could check if it's in `pytz.all_timezones_set` instead of catching the exception - OTOH, it's a very rare case so maybe catching an exception for that is cleaner?

Reply:
in the end I removed `as_timezone` altogether since it was always True